tests/test_log.py

```diff
@@ -1,6 +1,39 @@
+import logging
+
 import pytest
 
-from mandible.log import log_errors
+from mandible.log import init_custom_log_record_factory, log_errors
+
+
+@pytest.fixture
+def event_config():
+    return {
+        "cma": {
+            "event": {
+                "cumulus_meta": {
+                    "execution_name": "this_is_a_test_execution",
+                    "cumulus_version": "TEST_CUMULUS"
+                },
+                "payload": {
+                    "granules": [{"granuleId": "test_granule"}]
+                }
+            }
+        }
+    }
+
+
+def test_custom_log_record_factory(caplog, monkeypatch, event_config):
+    monkeypatch.setenv("DAAC_VERSION", "TEST_DAAC")
+    monkeypatch.setenv("CORE_VERSION", "TEST_CORE")
+    caplog.set_level(logging.INFO)
+    log = logging.getLogger()
+    init_custom_log_record_factory(event_config)
+    log.info("TEST")
+    assert "this_is_a_test_execution" in caplog.text
+    assert "TEST_CUMULUS" in caplog.text
+    assert "TEST_DAAC" in caplog.text
+    assert "TEST_CORE" in caplog.text
+    assert "test_granule" in caplog.text
```

Comment:
IMO, use `caplog.records[0].attribute_name`, since we're not doing anything with formatters here. The assertions are also pretty weak because they're not checking that the values are showing up in the right places, they would still pass if there were copy/paste mistakes that lead the values to be added to the wrong attributes.

Alternatively if you defined the formatter within the test I might be more amenable to using `caplog.text` or `caplog.messages`. That would also make it easier to write an assert statement using `==` rather than `in` to ensure that stuff shows up in the correct order.

Reply:
I don't think I need to test that values are in any order. It's a matter of making sure the attributes are in the log record. I will take a look at the record[0] again